tests/functional/adapter/materialization/test_snapshot.py

```diff
@@ -0,0 +1,137 @@
+import pytest
+from dbt.tests.adapter.basic.test_snapshot_check_cols import BaseSnapshotCheckCols
+from dbt.tests.adapter.basic.test_snapshot_timestamp import BaseSnapshotTimestamp
+
+seeds_base_csv = """
+id,name,some_date
+1,Easton,1981-05-20 06:46:51
+2,Lillian,1978-09-03 18:10:33
+3,Jeremiah,1982-03-11 03:59:51
+4,Nolan,1976-05-06 20:21:35
+5,Hannah,1982-06-23 05:41:26
+6,Eleanor,1991-08-10 23:12:21
+7,Lily,1971-03-29 14:58:02
+8,Jonathan,1988-02-26 02:55:24
+9,Adrian,1994-02-09 13:14:23
+10,Nora,1976-03-01 16:51:39
+""".lstrip()
+
+
+seeds_added_csv = (
+    seeds_base_csv
+    + """
+11,Mateo,2014-09-07 17:04:27
+12,Julian,2000-02-04 11:48:30
+13,Gabriel,2001-07-10 07:32:52
+14,Isaac,2002-11-24 03:22:28
+15,Levi,2009-11-15 11:57:15
+16,Elizabeth,2005-04-09 03:50:11
+17,Grayson,2019-08-06 19:28:17
+18,Dylan,2014-03-01 11:50:41
+19,Jayden,2009-06-06 07:12:49
+20,Luke,2003-12-05 21:42:18
+""".lstrip()
+)
+
+seeds_newcolumns_csv = """
```

Comment:
what is this seed being used for?

Reply:
This seed is not used in the dbt provided test. It seems that the following behaviour is untested.

https://docs.getdbt.com/docs/building-a-dbt-project/snapshots#snapshot-configurations

When the columns of your source query changes, dbt will attempt to reconcile this change in the destination snapshot [table](https://docs.getdbt.com/terms/table). dbt does this by:

> Creating new columns from the source query in the destination table
> Expanding the size of string types where necessary (eg. varchars on Redshift)
> dbt will not delete columns in the destination snapshot table if they are removed from the source query. It will also not change the type of a column beyond expanding the size of varchar columns. That is, if a string column is changed to a date column in the snapshot source query, dbt will not attempt to change the type of the column in the destination table.

I will see if i can make that work.

Reply:
I added the test, based on my reading of the dbt docs. it doesn't update the existing data.  See https://docs.getdbt.com/docs/faqs/snapshot-schema-changes